plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-order-refunds-v2-controller.php

```diff
@@ -320,7 +322,20 @@ protected function prepare_object_for_database( $request, $creating = false ) {
 		 * @param WP_REST_Request $request  Request object.
 		 * @param bool            $creating If is creating a new object.
 		 */
-		return apply_filters( "woocommerce_rest_pre_insert_{$this->post_type}_object", $refund, $request, $creating );
+		$refund = apply_filters( "woocommerce_rest_pre_insert_{$this->post_type}_object", $refund, $request, $creating );
+
+		// If the filtered result is not a WC_Data instance and is not a WP_Error then something went wrong, but we
+		// still need to honor the declared return type.
+		return Types::ensure_instance_of(
+			$refund,
+			WC_Data::class,
```

Comment:
```suggestion
			WC_Order_Refund::class,
```

This seems pretty generic. Should this be `WC_Order_Refund` instead?

Reply:
I think that might make sense, certainly that's the datatype we'd expect to see flowing through here in most out-of-the-box scenarios, but if so we should also update the `@return` tag for the method itself, which is currently documented as returning either a `WC_Data` or `WP_Error` object.

We would need some careful consideration before doing that, however. For instance, hypothetically, could there be cases where someone has defined `My_Custom_Refund` as a subclass of `WC_Order` or `WC_Abstract_Order`? If so, narrowing the type to `WC_Order_Refund` in this way would cause breakages, etc.

Reply:
So, coming back to this: I'm erring on the side of treading lightly, at least initially, and just going with the currently declared return type. However, please flag if you feel that isn't going far enough (I just worry narrowing the acceptable types introduces more risk).